src/passes/pass.cpp

```diff
@@ -138,19 +138,23 @@ void PassRunner::addDefaultGlobalOptimizationPasses() {
 }
 
 void PassRunner::run() {
-  if (options.debug) {
+  // BINARYEN_PASS_DEBUG is a convenient commandline way to log out the toplevel passes, their times,
+  //                     and validate between each pass.
+  //                     (we don't recurse pass debug into sub-passes, as it doesn't help anyhow and
+  //                     also is bad for e.g. printing which is a pass)
+  static const int passDebug = getenv("BINARYEN_PASS_DEBUG") ? atoi(getenv("BINARYEN_PASS_DEBUG")) : 0;
+  if (!isNested() && (options.debug || passDebug)) {
```

Comment:
Another thought I had was to split the debug run into a separate function from the non-debug run, so it's clearer that the debug run has its own weirdness that maybe we don't need to care about elsewhere.

How about:
* remove `isNested`
* make `PassRunner::run` the virtual method
* move the debug run logic to `doRunDebug` and the non-debug case to `doRun`, both protected non-virtual methods
* `PassRunner::run` can just be
```c++
if (options.debug || passDebug) {
  doRunDebug();
} else {
  doRun();
}
```
and `NestedPassRunner::run` can always just call `doRun()`

That's the more OO way to do it. Alternatively we can let `isNested` be a field on `PassRunner` and not need subclassing at all. Though if we have enough different logic, using dynamic dispatch instead of repeatedly checking `if (isNested)` is probably cleaner.
And either way I think it's worth it to split this out into `doRun` and `doRunDebug` at this point.